packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer_test.go

```diff
@@ -279,22 +330,1073 @@ func TestRoutingStarFullConnectivity(t *testing.T) {
+		FetchAITestKeys[1], AgentsTestAddresses[1], []string{peer.MultiAddr()},
+	)
+	if err != nil {
+		t.Fatal("Failed to initialize DHTClient:", err)
+	}
+	defer clientCleanup1()
+
+	client2, clientCleanup2, err := SetupDHTClient(
+		FetchAITestKeys[2], AgentsTestAddresses[2], []string{peer.MultiAddr()},
+	)
+	if err != nil {
+		t.Fatal("Failed to initialize DHTClient:", err)
+	}
+	defer clientCleanup2()
+
+	rxClient1 := make(chan *aea.Envelope)
+	client1.ProcessEnvelope(func(envel *aea.Envelope) error {
+		rxClient1 <- envel
+		return client1.RouteEnvelope(&aea.Envelope{
+			To:     envel.Sender,
+			Sender: envel.To,
+		})
+	})
+
+	rxClient2 := make(chan *aea.Envelope)
+	client2.ProcessEnvelope(func(envel *aea.Envelope) error {
+		rxClient2 <- envel
+		return nil
+	})
+
+	time.Sleep(1 * time.Second)
+	err = client2.RouteEnvelope(&aea.Envelope{
+		To:     AgentsTestAddresses[1],
+		Sender: AgentsTestAddresses[2],
+	})
+	if err != nil {
+		t.Error("Failed to RouteEnvelope from DHTClient to DHTClient:", err)
+	}
+
+	expectEnvelope(t, rxClient1)
+	expectEnvelope(t, rxClient2)
+
+}
+
+// TestRoutingDHTClientToDHTClientIndirect dht client to dht client connected to a different peer
+func TestRoutingDHTClientToDHTClientIndirect(t *testing.T) {
+	peer1, peerCleanup1, err := SetupLocalDHTPeer(
+		FetchAITestKeys[0], AgentsTestAddresses[0], DefaultLocalPort, DefaultDelegatePort,
+		[]string{},
+	)
+	if err != nil {
+		t.Fatal("Failed to initialize DHTPeer:", err)
+	}
+	defer peerCleanup1()
+
+	peer2, peerCleanup2, err := SetupLocalDHTPeer(
+		FetchAITestKeys[1], AgentsTestAddresses[1], DefaultLocalPort+1, DefaultDelegatePort+1,
+		[]string{peer1.MultiAddr()},
+	)
+	if err != nil {
+		t.Fatal("Failed to initialize DHTPeer:", err)
+	}
+	defer peerCleanup2()
+
+	client1, clientCleanup1, err := SetupDHTClient(
+		FetchAITestKeys[2], AgentsTestAddresses[2], []string{peer1.MultiAddr()},
+	)
+	if err != nil {
+		t.Fatal("Failed to initialize DHTClient:", err)
+	}
+	defer clientCleanup1()
+
+	client2, clientCleanup2, err := SetupDHTClient(
+		FetchAITestKeys[3], AgentsTestAddresses[3], []string{peer2.MultiAddr()},
+	)
+	if err != nil {
+		t.Fatal("Failed to initialize DHTClient:", err)
+	}
+	defer clientCleanup2()
+
+	rxClient1 := make(chan *aea.Envelope)
+	client1.ProcessEnvelope(func(envel *aea.Envelope) error {
+		rxClient1 <- envel
+		return client1.RouteEnvelope(&aea.Envelope{
+			To:     envel.Sender,
+			Sender: envel.To,
+		})
+	})
+
+	rxClient2 := make(chan *aea.Envelope)
+	client2.ProcessEnvelope(func(envel *aea.Envelope) error {
+		rxClient2 <- envel
+		return nil
+	})
+
+	time.Sleep(1 * time.Second)
+	err = client2.RouteEnvelope(&aea.Envelope{
+		To:     AgentsTestAddresses[2],
+		Sender: AgentsTestAddresses[3],
+	})
+	if err != nil {
+		t.Error("Failed to RouteEnvelope from DHTClient to DHTClient:", err)
+	}
+
+	expectEnvelope(t, rxClient1)
+	expectEnvelope(t, rxClient2)
+
+}
+
+/*
+	DHT network: DelegateClient
```

Comment:
Tests different patterns of `DelegateClient` to {`DHTPeer`, `DHTClient`, `DelegateClient`} communications